src/array.jl

```diff
@@ -319,6 +319,9 @@ Adapt.adapt_storage(::Float32Adaptor, xs::AbstractArray) =
 Adapt.adapt_storage(::Float32Adaptor, xs::AbstractArray{<:AbstractFloat}) =
   isbits(xs) ? xs : convert(CuArray{Float32}, xs)
 
+Adapt.adapt_storage(::Float32Adaptor, xs::AbstractArray{<:Complex{<:AbstractFloat}}) =
+  isbits(xs) ? xs : convert(CuArray{Float32}, xs)
```

Comment:
Does this work?

```
julia> convert(CuArray{Float32}, ComplexF64[1])
1-element CuArray{Float32,1}:
 1.0

julia> convert(CuArray{ComplexF32}, ComplexF64[1])
1-element CuArray{Complex{Float32},1}:
 1.0f0 + 0.0f0im
```

Reply:
Both of those work as expected.

Reply:
What I meant is that you're calling `convert(CuArray{Float32}, xs)` for arguments of eltype `Complex{<:AbstractFloat}`, which results in:

```
julia> cu(ComplexF64[1])
1-element CuArray{Float32,1}:
 1.0

julia> cu(ComplexF64[1+1im])
ERROR: InexactError: Float32(1.0 + 1.0im)
```

Reply:
Oh, that's a typo, I had the correct definition on the REPL, when I was checking this worked. Should be fixed now.